core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,628 @@ def update_question_suggestion(
+
+
+def _create_question_review_stats_from_models(
+    question_review_stats_model):
+    """Creates a domain object representing the supplied
+    QuestionReviewStatsModel.
+
+    Args:
+        question_review_stats_model: QuestionReviewStatsModel.
+            The model to convert to a domain object.
+
+    Returns:
+        QuestionReviewStats. The corresponding QuestionReviewStats domain
+        object.
+    """
+    return suggestion_registry.QuestionReviewStats(
+        question_review_stats_model.reviewer_user_id,
+        question_review_stats_model.topic_id,
+        question_review_stats_model.reviewed_questions_count,
+        question_review_stats_model.accepted_questions_count,
+        question_review_stats_model.accepted_questions_with_reviewer_edits_count,
+        question_review_stats_model.first_contribution_date,
+        question_review_stats_model.last_contribution_date
+    )
+
+
+def _get_all_translation_review_stats(user_id):
+    """Gets all TranslationReviewStatsModels corresponding to the supplied
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(TranslationReviewStats). TranslationReviewStats domain objects
+        corresponding to the supplied user.
+    """
+    translation_review_stats_models = (
+        suggestion_models.TranslationReviewStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_translation_review_stats_from_models(model)
+        for model in translation_review_stats_models
+    ]
+
+
+def _get_all_question_contribution_stats(user_id):
+    """Gets all QuestionContributionStatsModels corresponding to the supplied
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(QuestionContributionStats). QuestionContributionStats domain objects
+        corresponding to the supplied user.
+    """
+    question_contribution_stats_models = (
+        suggestion_models.QuestionContributionStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_question_contribution_stats_from_models(model)
+        for model in question_contribution_stats_models
+    ]
+
+
+def _get_all_question_review_stats(user_id):
+    """Gets all QuestionReviewStatsModels corresponding to the supplied
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(QuestionReviewStats). QuestionReviewStats domain objects
+        corresponding to the supplied user.
+    """
+    question_review_stats_models = (
+        suggestion_models.QuestionReviewStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_question_review_stats_from_models(model)
+        for model in question_review_stats_models
+    ]
+
+
+def get_all_contributor_stats(user_id):
```

Comment:
Just to check -- these are four rather heavy GET requests. Will you need them all at once in the frontend?

Reply:
Yes we fetch them once https://docs.google.com/document/d/16SciF9u-AngOdmvBaSyjYCAYrppR6lTE-S2vwhhh86A/edit#bookmark=id.rpcmy3f6cyjv